tools/devops/automation/templates/windows/build.yml

```diff
@@ -69,6 +69,32 @@ steps:
   displayName: 'Show .NET info'
   continueOnError: true
 
+- task: AzureKeyVault@2
+  inputs:
+    azureSubscription: 'Xamarin - R&D - XamarinSecurity'
+    KeyVaultName: 'xamarin-ios-vault'
+    SecretsFilter: 'RemoteMacIdRsa'
+  displayName: 'Download id_rsa'
+
+- pwsh: |
+    $idRsaPath = "$(Get-Location)\id_rsa"
+    Write-Host "##vso[task.setvariable variable=ID_RSA_PATH]$idRsaPath"
+    Add-Content -Path "id_rsa"  -Value "$(RemoteMacIdRsa)"
+    # We need to make sure the private key is only accessible by the current user,
+    # otherwise ssh will complain and not use it.
+    icacls id_rsa /inheritance:r
+    $grant="$Env:USERNAME" + ":(R)"
```

Comment:
You can do the following in pwsh:

```pwsh
$grant="$($Env:USERNAME):(R)"
```
Will to the string replacement that you want.